pkg/workflow/providers/oam/apply.go

```diff
@@ -91,21 +90,17 @@ func (p *provider) RenderComponent(ctx wfContext.Context, v *value.Value, act wf
 	return nil
 }
 
-func (p *provider) applyComponent(_ wfContext.Context, v *value.Value, act wfTypes.Action, mu *sync.Mutex) error {
-	comp, patcher, clusterName, overrideNamespace, env, err := lookUpValues(v, mu)
+// ApplyComponent apply component.
+func (p *provider) ApplyComponent(ctx wfContext.Context, v *value.Value, act wfTypes.Action) error {
+	comp, patcher, clusterName, overrideNamespace, env, err := lookUpCompInfo(v)
 	if err != nil {
 		return err
 	}
-	workload, traits, healthy, err := p.apply(*comp, patcher, clusterName, overrideNamespace, env)
+	workload, traits, healthy, err := p.apply(*comp, patcher, clusterName, overrideNamespace, "", env)
 	if err != nil {
 		return err
 	}
 
-	if mu != nil {
```

Comment:
Why remove this? It is used to prevent race problem.

Reply:
All callers pass a nil to this param. It is not actually used.

Reply:
lgtm